File: Java/Program/Komputer.java
import java.util.ArrayList;
import java.util.List;

public class Komputer extends ElectronicDevice {
    protected String nama;
    protected Cpu cpu;
    protected Motherboard motherboard;
    protected List<Ram> ramList;
    protected List<Harddrive> harddriveList;
    protected PowerSupply powerSupply;
    protected List<GraphicsCard> gpuList;
    protected List<Peripheral> peripherals;
    protected String connectionType;
    protected String coolingType;

    public Komputer() {
        super();
        this.ramList = new ArrayList<>();
        this.harddriveList = new ArrayList<>();
        this.gpuList = new ArrayList<>();
        this.peripherals = new ArrayList<>();
    }

    public Komputer(String nama, Cpu cpu, Motherboard motherboard, List<Ram> ramList, 
                    List<Harddrive> harddriveList, PowerSupply powerSupply, List<GraphicsCard> gpuList, List<Peripheral> peripherals, String connectionType, String coolingType,
                    String powerSource, int powerConsumptionWatt) {
        super(powerSource, powerConsumptionWatt);
        this.nama = nama;
        this.cpu = cpu;
        this.motherboard = motherboard;
        this.ramList = ramList;
        this.harddriveList = harddriveList;
        this.powerSupply = powerSupply;
        this.gpuList = gpuList;
        this.peripherals = peripherals;
        this.connectionType = connectionType;
        this.coolingType = coolingType;
    }

    public void setNama(String nama) {
        this.nama = nama;
    }

    public String getNama() {
        return this.nama;
    }

    public void setCpu(Cpu cpu) {
        this.cpu = cpu;
    }

    public Cpu getCpu() {
        return this.cpu;
    }

    public void setMotherboard(Motherboard motherboard) {
        this.motherboard = motherboard;
    }

    public Motherboard getMotherboard() {
        return this.motherboard;
    }

    public void setRamList(List<Ram> ramList) {
        this.ramList = ramList;
    }

    public List<Ram> getRamList() {
        return this.ramList;
    }

    public void setHarddriveList(List<Harddrive> harddriveList) {
        this.harddriveList = harddriveList;
    }

    public List<Harddrive> getHarddriveList() {
        return this.harddriveList;
    }

    public void setPowerSupply(PowerSupply powerSupply) {
        this.powerSupply = powerSupply;
    }

    public PowerSupply getPowerSupply() {
        return this.powerSupply;
    }

    public void setGpuList(List<GraphicsCard> gpuList) {
        this.gpuList = gpuList;
    }

    public List<GraphicsCard> getGpuList() {
        return this.gpuList;
    }

    public void setPeripherals(List<Peripheral> peripherals) {
        this.peripherals = peripherals;
    }

    public List<Peripheral> getPeripherals() {
        return this.peripherals;
    }

    public void addRam(Ram ram) {
        this.ramList.add(ram);
    }

    public void addHarddrive(Harddrive harddrive) {
        this.harddriveList.add(harddrive);
    }

    public void addGraphicsCard(GraphicsCard gpu) {
        this.gpuList.add(gpu);
    }

    public void addPeripheral(Peripheral peripheral) {
        this.peripherals.add(peripheral);
    }

    public int getTotalRamCapacity() {
        int total = 0;
        for (Ram ram : ramList) {
            total += ram.getKapasitasGB();
        }
        return total;
    }

    public int getTotalStorageCapacity() {
        int total = 0;
        for (Harddrive hd : harddriveList) {
            total += hd.getKapasitasGB();
        }
        return total;
    }

    public void setConnectionType(String connectionType) {
        this.connectionType = connectionType;
    }

    public String getConnectionType() {
        return this.connectionType;
    }

    public void setCoolingType(String coolingType) {
        this.coolingType = coolingType;
    }

    public String getCoolingType() {
        return this.coolingType;
    }
}

File: Python/Program/Komputer.py
from ElectronicDevice import ElectronicDevice
from Cpu import Cpu
from Harddrive import Harddrive
from Ram import Ram
from Motherboard import Motherboard
from PowerSupply import PowerSupply
from GraphicsCard import GraphicsCard
from Peripheral import Peripheral

class Komputer(ElectronicDevice):
    def __init__(self, nama="", cpu=None, motherboard=None, ram_list=None, harddrive_list=None, power_supply=None, gpu_list=None, peripherals=None, connection_type="", cooling_type="", power_source="", power_consumption_watt=0):
        super().__init__(power_source, power_consumption_watt)
        self.nama = nama
        self.cpu = cpu if cpu is not None else Cpu()
        self.motherboard = motherboard if motherboard is not None else Motherboard()
        self.ram_list = ram_list if ram_list is not None else []
        self.harddrive_list = harddrive_list if harddrive_list is not None else []
        self.power_supply = power_supply if power_supply is not None else PowerSupply()
        self.gpu_list = gpu_list if gpu_list is not None else []
        self.peripherals = peripherals if peripherals is not None else []
        self.connection_type = connection_type
        self.cooling_type = cooling_type

    def set_nama(self, nama):
        self.nama = nama

    def get_nama(self):
        return self.nama

    def set_cpu(self, cpu):
        self.cpu = cpu

    def get_cpu(self):
        return self.cpu

    def set_motherboard(self, motherboard):
        self.motherboard = motherboard

    def get_motherboard(self):
        return self.motherboard

    def set_ram_list(self, ram_list):
        self.ram_list = ram_list

    def get_ram_list(self):
        return self.ram_list

    def set_harddrive_list(self, harddrive_list):
        self.harddrive_list = harddrive_list

    def get_harddrive_list(self):
        return self.harddrive_list

    def set_power_supply(self, power_supply):
        self.power_supply = power_supply

    def get_power_supply(self):
        return self.power_supply

    def set_gpu_list(self, gpu_list):
        self.gpu_list = gpu_list

    def get_gpu_list(self):
        return self.gpu_list

    def add_ram(self, ram):
        self.ram_list.append(ram)

    def add_harddrive(self, harddrive):
        self.harddrive_list.append(harddrive)

    def add_graphics_card(self, gpu):
        self.gpu_list.append(gpu)

    def add_peripheral(self, peripheral):
        self.peripherals.append(peripheral)

    def get_peripherals(self):
        return self.peripherals

    def get_total_ram_capacity(self):
        return sum(ram.get_kapasitas_gb() for ram in self.ram_list)

    def get_total_storage_capacity(self):
        return sum(hd.get_kapasitas_gb() for hd in self.harddrive_list)

    def set_connection_type(self, connection_type):
        self.connection_type = connection_type

    def get_connection_type(self):
        return self.connection_type

    def set_cooling_type(self, cooling_type):
        self.cooling_type = cooling_type

    def get_cooling_type(self):
        return self.cooling_type